### PR TITLE
folly v2023.07.03

### PR DESCRIPTION
**Pull request by jj683**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/f/folly/xmake.lua [hunk 1]
```diff
@@ -43,6 +44,9 @@ package("folly")
         if package:is_plat("windows") then
             table.insert(configs, "-DBoost_USE_STATIC_RUNTIME=" .. (package:config("vs_runtime"):startswith("MT") and "ON" or "OFF"))
         end
+        if package:config("no_exception_tracer") then
+            table.insert(configs, "-DFOLLY_NO_EXCEPTION_TRACER=ON")
+        end
```
Comment (jj683) [inline packages/f/folly/xmake.lua:49]: https://github.com/facebook/folly/blob/b88123c2abf4b3244ed285e6db0d4bea2d24f95f/CMakeLists.txt#L193

File: packages/f/folly/xmake.lua [hunk 2]
```diff
@@ -12,8 +12,9 @@ package("folly")
     add_versions("2022.02.14", "6a50d4cc954f0f77efc85af231ee6b7f676a0d743c12b7080fb03fd3da3ffbf2")
     add_versions("2022.04.25", "7d5cd59613780d7d2b9c946c810bfd6b0d11ed3a8a74c5ab00d4e9de5d1b2104")
     add_versions("2022.08.29", "3adac6d4b203c2917185fd190fc04d615051cb2a4f6b988ddf3c42034efc8d4d")
+    add_versions("2023.07.03", "e49901ac87bca18c0a2f71af340d63f5573bae463fdeb0c32092c6bf45621474")
 
-    add_patches("<=2022.08.29", path.join(os.scriptdir(), "patches", "2021.06.28", "reorder.patch"), "9a6bf283881580474040cfc7a8e89d461d68b89bae5583d89fff0a3198739980")
+    add_patches("<=2023.07.03", path.join(os.scriptdir(), "patches", "2021.06.28", "reorder.patch"), "9a6bf283881580474040cfc7a8e89d461d68b89bae5583d89fff0a3198739980")
     add_patches("<=2022.08.29", path.join(os.scriptdir(), "patches", "2021.06.28", "regex.patch"), "6a77ade9f48dd9966d3f7154e66ca8a5c030ae2b6d335cbe3315784aefd8f495")
```
Comment (jj683) [inline packages/f/folly/xmake.lua:18]: Seems like they fixed problems with windows builds in master:
https://github.com/facebook/folly/commit/da3babcd57495a1e9db06afcefceff0bb13acf4a

**Comment by jj683** [comment]
Hm, even version from master (2022.08.29) have same problems with Windows & CMake. Let me check previous CMake version.

File: packages/f/folly/xmake.lua [hunk 3]
```diff
@@ -12,15 +12,18 @@ package("folly")
...
 
+    add_configs("no_exception_tracer", {description = "Exclude exception tracer, which is necessary to statically link libstdc++.", default = false, type = "boolean"})
+
     if is_plat("windows") then
         add_configs("shared", {description = "Build shared library.", default = false, type = "boolean", readonly = true})
     end
 
-    add_deps("cmake")
+    add_deps("cmake <=3.24.2")
```
Comment (waruqi) [inline packages/f/folly/xmake.lua:26]: why?

Comment (jj683) [reply packages/f/folly/xmake.lua]: I tried to investigate the reason why Windows builds failing with CMake errors, but they're failing even with master version:
https://github.com/xmake-io/xmake-repo/pull/2297/checks?check_run_id=14889087445

So this PR (config option + version update) make no degradation.

**Comment by waruqi** [comment]
maybe it's cmakelists.txt has some problems.

> 2023-07-09T16:55:55.1205230Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1206359Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1207160Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1207727Z   language, or location of object files:
2023-07-09T16:55:55.1208066Z 
2023-07-09T16:55:55.1208473Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1208929Z 
2023-07-09T16:55:55.1209056Z 
2023-07-09T16:55:55.1209314Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1209894Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1210500Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1211001Z   language, or location of object files:
2023-07-09T16:55:55.1211326Z 
2023-07-09T16:55:55.1211719Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1212154Z 
2023-07-09T16:55:55.1212686Z 
2023-07-09T16:55:55.1213291Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1213824Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1214372Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1215208Z   language, or location of object files:
2023-07-09T16:55:55.1215503Z 
2023-07-09T16:55:55.1215861Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1216234Z 
2023-07-09T16:55:55.1216344Z 
2023-07-09T16:55:55.1216563Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1217018Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1217542Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1217981Z   language, or location of object files:
2023-07-09T16:55:55.1218262Z 
2023-07-09T16:55:55.1218606Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1219112Z 
2023-07-09T16:55:55.1219222Z 
2023-07-09T16:55:55.1219473Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1219962Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1220491Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1220941Z   language, or location of object files:
2023-07-09T16:55:55.1221208Z 
2023-07-09T16:55:55.1221543Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1221901Z 
2023-07-09T16:55:55.1222003Z 
2023-07-09T16:55:55.1222210Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1222636Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1223146Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1223583Z   language, or location of object files:
2023-07-09T16:55:55.1223851Z 
2023-07-09T16:55:55.1224363Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1225010Z 
2023-07-09T16:55:55.1225113Z 
2023-07-09T16:55:55.1225498Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1225975Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1226479Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1227192Z   language, or location of object files:
2023-07-09T16:55:55.1227497Z 
2023-07-09T16:55:55.1227815Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1228319Z 
2023-07-09T16:55:55.1228439Z 
2023-07-09T16:55:55.1228906Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1229506Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1230010Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1230448Z   language, or location of object files:
2023-07-09T16:55:55.1230718Z 
2023-07-09T16:55:55.1231029Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1231393Z 
2023-07-09T16:55:55.1231493Z 
2023-07-09T16:55:55.1231730Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1232182Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1232683Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1233120Z   language, or location of object files:
2023-07-09T16:55:55.1233368Z 
2023-07-09T16:55:55.1233695Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1234049Z 
2023-07-09T16:55:55.1234149Z 
2023-07-09T16:55:55.1234353Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1234791Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1235287Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1235709Z   language, or location of object files:
2023-07-09T16:55:55.1236069Z 
2023-07-09T16:55:55.1236405Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1236755Z 
2023-07-09T16:55:55.1236854Z 
2023-07-09T16:55:55.1237059Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1237501Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1238000Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1238419Z   language, or location of object files:
2023-07-09T16:55:55.1238683Z 
2023-07-09T16:55:55.1239008Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1239436Z 
2023-07-09T16:55:55.1239537Z 
2023-07-09T16:55:55.1239742Z CMake Error in CMakeLists.txt:
2023-07-09T16:55:55.1240208Z   Evaluation file to be written multiple times with different content.  This
2023-07-09T16:55:55.1241341Z   is generally caused by the content evaluating the configuration type,
2023-07-09T16:55:55.1241773Z   language, or location of object files:
2023-07-09T16:55:55.1242040Z 
2023-07-09T16:55:55.1242368Z    C:/Users/runneradmin/AppData/Local/.xmake/cache/packages/2307/f/folly/2023.07.03/source/build_bf16b696/libfolly.pc
2023-07-09T16:55:55.1242721Z

**Comment by jj683** [comment]
Seems like some degradation happened due to Windows CI image update (vcpkg version or something), because even previous version of folly from master with old xmake and xmake-repo versions have the same errors:
https://github.com/jj683/xmake-repo/actions/runs/5518290500/jobs/10062039517

Because of that, folly even from xmake-report@master can't pass CI.

I'll try to figure out the problem with vcpkg files generation from folly cmake, but the easiest fix is just to disable this failing vcpkg files generator.

**Pull request closed without merging**